apps/proxy-auth/src/app/otp/component/login/login.component.ts

```diff
@@ -0,0 +1,113 @@
+import { Component, OnInit } from '@angular/core';
+import { LoginComponentStore } from './login.store';
+import { Observable, takeUntil } from 'rxjs';
+import { IAppState } from '../../store/app.state';
+import { select, Store } from '@ngrx/store';
+import { selectWidgetData } from '../../store/selectors';
+import { BaseComponent } from '@proxy/ui/base-component';
+import { FeatureServiceIds } from '@proxy/models/features-model';
+import { FormControl, FormGroup, Validators } from '@angular/forms';
+import { IlogInData, IOtpData, IResetPassword } from '../../model/otp';
+import { EMAIL_REGEX, PASSWORD_REGEX } from '@proxy/regex';
+import { CustomValidators } from '@proxy/custom-validator';
+
+@Component({
+    selector: 'proxy-login',
+    templateUrl: './login.component.html',
+    styleUrls: ['./login.component.scss'],
+    providers: [LoginComponentStore],
+})
+export class LoginComponent extends BaseComponent implements OnInit {
+    public state: string;
+    public step: number = 1;
+    public selectWidgetData$: Observable<any>;
+    public otpData$: Observable<any> = this.componentStore.otpdata$;
+    public isLoading$: Observable<boolean> = this.componentStore.isLoading$;
+    public resetPassword$: Observable<any> = this.componentStore.resetPassword$;
+    public loginForm = new FormGroup({
+        username: new FormControl<string>(null, [Validators.required, Validators.pattern(EMAIL_REGEX)]),
+        password: new FormControl<string>(null, [Validators.required]),
+    });
+    public emailForm = new FormGroup({
+        email: new FormControl<string>(null, [Validators.required]),
+    });
+    public resetPasswordForm = new FormGroup({
+        email: new FormControl<string | null>({ value: null, disabled: true }, [
```

Comment:
why are you managing email twice one in emailForm and second in resetPasswordForm
and type will be string only